snap/snapcraft.yaml

```diff
@@ -15,43 +16,46 @@ apps:
         environment:
           XDG_DATA_HOME: "$SNAP/usr/share"
           FONTCONFIG_PATH: "$SNAP/etc/fonts"
-          JAVA_HOME: "$SNAP/usr/lib/jvm/java-11-openjdk"
+          JAVA_HOME: "$SNAP/usr/lib/jvm/java-17-openjdk"
           LD_LIBRARY_PATH: "$SNAP/usr/lib/$SNAPCRAFT_ARCH_TRIPLET"
-        command: |
-            usr/lib/jvm/java-11-openjdk/bin/java -DJENKINS_HOME="$SNAP_DATA" -Djava.awt.headless=true -jar $SNAP/jenkins.war
+        command: launch
+            
         daemon: simple
     config:
         command: jenkins-config
 
 parts:
-    bsi-trigger: # A non-built part, only used to trigger builds in build.snapcraft.io on upstream changes
-        plugin: nil
-        source: https://github.com/jenkinsci/jenkins.git
     jenkins:
         plugin: nil
         stage-packages:
             - fonts-dejavu-core
             - libfontconfig1
-            - openjdk-11-jre-headless
+            - openjdk-17-jre-headless
             - ca-certificates-java
         build-packages:
             - curl
             - unzip
             - libnss3
             - ca-certificates-java
+        
         override-build: |
             curl -sLO http://mirrors.jenkins.io/war/latest/jenkins.war
             unzip jenkins.war META-INF/MANIFEST.MF
             VERSION=$(grep Jenkins-Version META-INF/MANIFEST.MF | cut -d' ' -f2 | tr -d '\r')
             rm META-INF/MANIFEST.MF
-            mv $SNAPCRAFT_PART_INSTALL/usr/lib/jvm/java-11-openjdk-* $SNAPCRAFT_PART_INSTALL/usr/lib/jvm/java-11-openjdk
-            rm $SNAPCRAFT_PART_INSTALL/usr/lib/jvm/java-11-openjdk/lib/security/cacerts
-            cp /etc/ssl/certs/java/cacerts $SNAPCRAFT_PART_INSTALL/usr/lib/jvm/java-11-openjdk/lib/security/cacerts
+            mv $SNAPCRAFT_PART_INSTALL/usr/lib/jvm/java-17-openjdk-* $SNAPCRAFT_PART_INSTALL/usr/lib/jvm/java-17-openjdk
+            rm $SNAPCRAFT_PART_INSTALL/usr/lib/jvm/java-17-openjdk/lib/security/cacerts
+            cp /etc/ssl/certs/java/cacerts $SNAPCRAFT_PART_INSTALL/usr/lib/jvm/java-17-openjdk/lib/security/cacerts
             cp /etc/ssl/certs/java/cacerts $SNAPCRAFT_PART_INSTALL/etc/ssl/certs/java/cacerts
             mv jenkins.war $SNAPCRAFT_PART_INSTALL
             snapcraftctl set-version "$VERSION"
+                       
     config:
         plugin: dump
         source: snap/local
         prime:
             - jenkins-config
+    launch:
+        plugin: dump
+        source: snap/scripts
```

Comment:
We should put the scripts in snap/local, so snap/local/bin/launch for example, so it would end up in $SNAP/bin/launch, and would no longer get this message:

```
                                                                                                                                                              
The 'snap' directory is meant specifically for snapcraft, but it contains the following non-snapcraft-related paths, which is unsupported and will cause unexpected behavior:                              
- scripts                                                                                                                                                                                                  
- scripts/launch                                                                                                                                                                                           
                                                                                                                                                                                                           
If you must store these files within the 'snap' directory, move them to 'snap/local', which is ignored by snapcraft.  
``

Reply:
Thanks @popey , I updated the yaml. See my latest push.

Reply:
Hey @popey , I forgot to add the fact that I removed the BGI trigger, as it didn't seem to be working as intended. There weren't any new builds in the `edge` channel, spurring this PR. 

I am working on a solution in my `testing` branch in my repo to see if I can get something working.